### PR TITLE
School Library

### PR DESCRIPTION
**Pull request by Zel-hub7**
1. I created class Person with their specific requirements
2. I created class Student from the person class.
3. I craeted teachers class from Person class.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class Person
+  attr_accessor :name
+  attr_reader :id, :age
+
```
Comment (iambenkis) [inline person.rb]: - [ ] Good job creating the Person class , 👏 . As for this project requirements , the `age` should be able to be modified , and read. Kindly define it as `attr_accessor` to meet the requirement 👍

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  def initialize(name = 'Unknown', age = nil, parent_permission: true)
+    @id = generate_id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```
Comment (iambenkis) [inline person.rb]: - [ ] Good job setting default value for optional variable 👍 , As for the `age` variable , it should not be defined with a default value as it is not a optional variable  , Kindly fix that 👍

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Student < Person
+  attr_reader :classroom
+
+  def initialize(name: 'Unknown', age: nil, parent_permission: true, classroom: 'No Classroom')
+    super(name: name, age: age, parent_permission: parent_permission)
+    @classroom = classroom
+  end
```
Comment (iambenkis) [inline student.rb]: - [ ] As I suggested in the previous comments  ,   `age` variable , it should not be defined with a default value as it is not a optional variable. For the student class ,    `classroom` should not be an optional varibale as well, Kindly fix that 😄

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+require_relative 'Person'
+
+class Teacher < Person
+  attr_reader :specialization
+
+  def initialize(name: 'Unknown', age: nil, parent_permission: true, specialization: 'General')
+    super(name: name, age: age, parent_permission: parent_permission)
+    @specialization = specialization
+  end
```
Comment (iambenkis) [inline teacher.rb]: - [ ]  Kindly apply the previous comment to the  `specialization`  variable😄

**Review by iambenkis — CHANGES_REQUESTED** [review]
#### Hi @Zel-hub7 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️  Person class added
✔️ Student class added
✔️ Teacher class added

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: person.rb (hunk 1 above)
Comment (AbelG101) [inline person.rb]: - [ ] As mentioned previously the `age` attribute should be able to be modified so kindly make sure to specify it with an `attr_accessor` along the `:name` attribute at line `2`.

Comment (AbelG101) [inline person.rb]: - [ ] As mentioned previously the `age` attribute should be able to be modified so kindly make sure to specify it with an `attr_accessor` along the `:name` attribute at line `2`.

File: student.rb (hunk 3 above)
Comment (AbelG101) [inline student.rb]: - [ ] Once again as mentioned previously the `age` variable in the constructor above shouldn't have a default value of `nil` as it should be specified whenever an instance of a student class is created so please make sure to remove it.

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @Zel-hub7 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Correct use of Git Flow ✔
- Linter check is passing ✔
- Good PR summary ✔

 ## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 3 above)
Comment (Zel-hub7) [reply student.rb]: Kindly I didn't understand this one...

**Review by mikethreels — APPROVED** [review]
Hi Zelalem, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job implementing the requested changes.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**